velox/functions/prestosql/types/TimestampWithTimeZoneType.cpp

```diff
@@ -109,26 +109,27 @@ void castToString(
   auto* flatResult = result.as<FlatVector<StringView>>();
   const auto* timestamps = input.as<SimpleVector<int64_t>>();
 
-  auto formatter =
-      functions::buildJodaDateTimeFormatter("yyyy-MM-dd HH:mm:ss.SSS zzzz");
-
-  context.applyToSelectedNoThrow(rows, [&](auto row) {
-    const auto timestampWithTimezone = timestamps->valueAt(row);
-
-    const auto timestamp = unpackTimestampUtc(timestampWithTimezone);
-    const auto timeZoneId = unpackZoneKeyId(timestampWithTimezone);
-    const auto* timezonePtr = tz::locateZone(tz::getTimeZoneName(timeZoneId));
-
-    exec::StringWriter<false> result(flatResult, row);
-
-    const auto maxResultSize = formatter->maxResultSize(timezonePtr);
-    result.reserve(maxResultSize);
-    const auto resultSize =
-        formatter->format(timestamp, timezonePtr, maxResultSize, result.data());
-    result.resize(resultSize);
-
-    result.finalize();
-  });
+  functions::buildJodaDateTimeFormatter("yyyy-MM-dd HH:mm:ss.SSS zzzz")
+      .thenOrThrow([&](auto formatter) {
```

Comment:
Are we Ok throwing Unexpected<Error>::BadExpectedAccess here?

```
* Sometimes when using a continuation you would prefer an exception to be
 * thrown for a value-less Expected. For that you can use .thenOrThrow, as
 * follows:
 *
 * B b = someFunc()
 *     .thenOrThrow([](A a){return B(a);});
 *
 * The above call to thenOrThrow will invoke the lambda if the Expected returned
 * by someFunc() contains a value. Otherwise, it will throw an exception of type
 * Unexpected<Error>::BadExpectedAccess. If you prefer it throw an exception of
 * a different type, you can pass a second lambda to thenOrThrow:
 *
 * B b = someFunc()
 *     .thenOrThrow([](A a){return B(a);},
 *                  [](Error e) {throw MyException(e);});

```

Reply:
Thanks for catching that. I changed this to use `VELOX_CHECK`.